File: java/src/com/szklimowicz/organisms/plants/Plant.java
package com.szklimowicz.organisms.plants;

import com.szklimowicz.Field;
import com.szklimowicz.World;
import com.szklimowicz.organisms.Organism;

import java.util.concurrent.ThreadLocalRandom;

public abstract class Plant extends Organism {

    public Plant(int x, int y, World world) {
        super(x, y, world);
        setName("Plant");
        setStrength(0);
        setInitiative(0);
    }

    @Override
    public void action() {
        if (Math.random() >= 0.96) {
            int new_x = getX(), new_y = getY();
            int range = getRange();
            int dir = ThreadLocalRandom.current().nextInt(1, 4 + 1);

            switch (dir) {

                case LEFT:
                    new_x -= range;
                    break;
                case RIGHT:
                    new_x += range;
                    break;
                case UP:
                    new_y -= range;
                    break;
                case DOWN:
                    new_y += range;
                    break;
            }

            World world = getWorld();
            Class plant_class = this.getClass();
            if (world.inWorldBorders(new_x, new_y)) {
                if (world.getFields()[new_y][new_x].getOrganism() == null) {
                    try {
                        Organism org = (Organism) plant_class.getDeclaredConstructor(int.class, int.class, World.class).newInstance(new_x, new_y, world);
                        world.addNewOrganism(org);
                    } catch (Exception ex) {
                        ex.printStackTrace();
                    }
                }
            }
        }
    }
}

File: python/plant.py
from organism import Organism

from random import randint

class Plant(Organism):

    def __init__(self, x, y, world):
        super().__init__(x, y, world)
        self.name = "Plant"

    def action(self, null):

        dir = randint(1,100)

        new_x = self.x
        new_y = self.y

        if(dir == 1):
            new_y = self.y - self.moverange
        if(dir == 2):
            new_y = self.y + self.moverange
        if(dir == 3):
            new_x = self.x - self.moverange
        if(dir == 4):
            new_x = self.x + self.moverange
        
        if new_x != self.x or new_y != self.y:
            if self.world.inWorld(new_x, new_y):
                if self.world.organisms[new_y][new_x] == 0:
                    self.world.addNewOrganism(type(self), new_x, new_y)